python/ray/air/_internal/usage.py

```diff
@@ -0,0 +1,18 @@
+from typing import TYPE_CHECKING
+
+from ray._private.usage.usage_lib import TagKey, record_extra_usage_tag
+
+if TYPE_CHECKING:
+    from ray.train.trainer import BaseTrainer
+
+
+def tag_air_trainer(trainer: "BaseTrainer"):
+    from ray.train.trainer import BaseTrainer
+
+    assert isinstance(trainer, BaseTrainer)
+    module_path = trainer.__module__
+    if module_path.startswith("ray.train"):
+        result = f"{trainer.__class__.__name__}"
+    else:
+        result = "custom_trainer"
+    record_extra_usage_tag(TagKey.AIR_TRAINER, result)
```

Comment:
@jjyao is there a way to mock this call for unit tests?

Reply:
You don't need to mock. You can just call it and then verify that usage stats report has it. See `test_usage_stats.py` for examples.

Reply:
@jjyao 
I actually prefer to have a mock provided. 

checking something like
```
        gcs_client = gcs_utils.GcsClient(address=ctx.address_info["gcs_address"])
        wait_for_condition(
            lambda: "worker_crash_system_error"
            in ray_usage_lib.get_extra_usage_tags_to_report(gcs_client),
            timeout=4,
        )
``` 
seems bleaching unit test boundary for me.

Reply:
Yea, I think currently our tests are more like integration tests. For unit tests, you can mock it.